code/modules/surgery/organs/heart.dm

```diff
@@ -254,3 +254,192 @@
 		owner.heal_overall_damage(15, 15, 0, BODYPART_ORGANIC)
 		if(owner.reagents.get_reagent_amount(/datum/reagent/medicine/ephedrine) < 20)
 			owner.reagents.add_reagent(/datum/reagent/medicine/ephedrine, 10)
+
+
+
+
+/obj/item/organ/heart/ethereal
+	name = "Crystal core"
+	icon_state = "stomach-p" //Welp. At least it's more unique in functionaliy.
+	desc = "A crystal-like organ that functions similar to a heart for Ethereals, it can revive it's owner."
+
+	///Cooldown for the next time we can crystalize
+	COOLDOWN_DECLARE(crystalize_cooldown)
+	///Timer ID for when we will be crystalized, If not preparing this will be null.
+	var/crystalize_timer_id
+	///The current crystal the ethereal is in, if any
+	var/obj/structure/ethereal_crystal/current_crystal
+	///Damage taken during crystalization, resets after it ends
+	var/crystalization_process_damage = 0
+	///Color of the heart, is set by the species on gain
+	var/ethereal_color = "#9c3030"
+
+
+/obj/item/organ/heart/ethereal/Insert(mob/living/carbon/M, special = 0)
+	. = ..()
+	RegisterSignal(M, COMSIG_MOB_STATCHANGE, .proc/on_stat_change)
+	RegisterSignal(M, COMSIG_LIVING_POST_FULLY_HEAL, .proc/on_owner_fully_heal)
```

Comment:
add a call to on_stat_change() to the list of things that happen when an ethereal heart is inserted into a corpse, in case someone wants to revive a corpse by replacing its heart with an ethereal one.

Reply:
no, i dont want it to work like that.